src/main/js/controls/Bubble/BubbleSingleDataset.js

```diff
@@ -0,0 +1,200 @@
+"use strict";
+
+import constants, { BUBBLE } from "../../helpers/constants";
+import { prepareLabelShapeItem } from "../../helpers/label";
+import { reflowLegend } from "../../helpers/legend";
+import {
+    createRegion,
+    hideAllRegions,
+    translateRegion,
+    areRegionsIdentical
+} from "../../helpers/region";
+import styles from "../../helpers/styles";
+import utils from "../../helpers/utils";
+import {
+    clear,
+    clickHandler,
+    hoverHandler,
+    prepareLegendItems,
+    processDataPoints,
+    translateBubbleGraph,
+    getDataPointValues,
+    calculateValuesRange
+} from "./helpers/helpers";
+import { draw, drawBubbles } from "./helpers/helpersSingleDataset";
+import Bubble from "./Bubble";
+
+/**
+ * A Bubble graph is a graph used to represent a collection of data
+ * points connected by a Bubble along the X and Y Axis.
+ *
+ * Lifecycle functions include:
+ *  * Load
+ *  * Generate
+ *  * Unload
+ *  * Destroy
+ *
+ * @module BubbleSingleDataset
+ * @class BubbleSingleDataset
+ */
+class BubbleSingleDataset extends Bubble {
+    /**
+     * @inheritdoc
+     * @public
+     */
+    load(graph) {
+        if (graph.content.length > 1) {
+            console.warn("BubbleSingleDataset can only use one dataset.");
+            return this;
+        }
+
+        if (
+            utils.isUndefined(graph.content[0].config.color) &&
+            utils.isDefined(graph.content[0].config.palette) &&
+            graph.content[0].config.values.length >
+                BUBBLE.MAX_BUBBLES_SINGLE_DATASET
+        ) {
+            console.warn("Cannot plot more than 4 bubbles using shades.");
```

Comment:
Is it worth having a warning? or Should we log an error if the consumer provides more than one dataset?

Reply:
The graph will be blank if there is more than 4 bubbles as agreed with UX. But any sort of feedback would be helpful for a consumer to know why the graph isn't rendering. I can update this to an error if that's more appropriate.

Reply:
Execution get stopped if we have more than 4 bubbles right?

Reply:
Yes, it won't render the data if there are more than 4 bubbles.

Reply:
Given that the graph will not load when this happens, I think an error could be appropriate